test_files/conditional_type/conditional_type.ts

```diff
@@ -0,0 +1 @@
+export type Filter<T, U> = T extends U ? T : never;
```

Comment:
I was going to say add an example of substitution type, but if I read this correctly - https://github.com/Microsoft/TypeScript/blob/master/src/compiler/types.ts#L4051

they can only occur internally to conditional types, thus can't be tested.